app/uk/gov/hmrc/tai/model/domain/TaxCodeHistory.scala

```diff
@@ -19,19 +19,19 @@ package uk.gov.hmrc.tai.model.domain
 import org.joda.time.LocalDate
 import play.api.libs.functional.syntax._
 import play.api.libs.json.Reads._
-import play.api.libs.json.{JsPath, Reads}
+import play.api.libs.json.{JsPath, Json, Reads}
+
+case class TaxCodeHistory(previous: TaxCodeRecord, current: TaxCodeRecord){
+
+  val mostRecentTaxCodeChangeDate: LocalDate = current.startDate
```

Comment:
hmmmm, Im not 100% on if this needs to change. Naming is the hardest thing. 

If you call the method currentStartDate you are then tying the method name to the implementation which means if we ever changed the logic to get the `mostRecentTaxCodeChangeDate`, e.g. it was passed down from the backend instead of using current.startDate, we would have to rename the method and change where the method was used.

Or maybe the implementation should be tied because the most recent tax code change date will always be the current tax code start date.

Reply:
Fair enough 👍